lib/spack/spack/build_systems/intel.py

```diff
@@ -0,0 +1,191 @@
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+import os
+import xml.etree.ElementTree as ET
+
+from llnl.util.filesystem import join_path
+from spack.package import PackageBase, run_after
+from spack.util.executable import Executable
+
+
+def _valid_components():
+    """A generator that yields valid components."""
+
+    tree = ET.parse('pset/mediaconfig.xml')
+    root = tree.getroot()
+
+    components = root.findall('//Abbr')
+    for component in components:
+        yield component.text
+
+
+class IntelPackage(PackageBase):
+    """Specialized class for licensed Intel software.
+
+    This class provides two phases that can be overridden:
+
+    1. :py:meth:`~.IntelPackage.configure`
+    2. :py:meth:`~.IntelPackage.install`
+
+    They both have sensible defaults and for many packages the
+    only thing necessary will be to override ``setup_environment``
+    to set the appropriate environment variables.
+    """
+    #: Phases of an Intel package
+    phases = ['configure', 'install']
+
+    #: This attribute is used in UI queries that need to know the build
+    #: system base class
+    build_system_class = 'IntelPackage'
+
+    #: By default, we assume that all Intel software requires a license.
+    #: This can be overridden for packages that do not require a license.
+    license_required = True
+
+    #: Comment symbol used in the ``license.lic`` file
+    license_comment = '#'
+
+    #: Location where Intel searches for a license file
+    license_files = ['Licenses/license.lic']
+
+    #: Environment variables that Intel searches for a license file
+    license_vars = ['INTEL_LICENSE_FILE']
+
+    #: URL providing information on how to acquire a license key
+    license_url = 'https://software.intel.com/en-us/articles/intel-license-manager-faq'
+
+    #: By default, install 'ALL' components
+    _components = ['ALL']
+
+    @property
+    def components(self):
+        """Components of the package to install."""
+        return self._components
+
+    @components.setter
+    def components(self, requested_components):
+        """Packages like ``intel-parallel-studio`` contain dozens of
+        components, and some users may want to install a subset of these.
+        This setter allows you to customize which components get installed.
+
+        By default, installs 'ALL' components. To install only MKL and DAAL,
+        for example, you would use:
+
+        .. code-block:: python
+
+           components = ['mkl', 'daal']
+        """
+        matches = []
+
+        for valid in _valid_components():
+            for requested in requested_components:
+                if requested in valid:
+                    matches.append(valid)
+
+        self._components = matches
```

Comment:
I'm beginning to realize just how much I don't know about Python. I'm attempting to create a getter/setter for `components` in `IntelPackage` that subclasses could call (see `intel` package below). I was hoping that a subclass could do something like:
```python
components = ['mkl', 'daal']
```
and the setter would add all matching components to `self._components`, but when I try to install `intel` I find that it doesn't call the getter/setter at all. I know we're using class attributes and properties interchangeably in other places in Spack, but it seems like that's not possible here? Can anyone explain where I'm going wrong or if there's a better way to get what I want?